feat(rome_js_analyze): noControlCharactersInRegex

unvalley opened the pull request:
<!--
	Thanks for submitting a pull request!

	We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request.

	Once created, your PR will be automatically labeled according to changed files.

	Learn more about contributing: https://github.com/rome/tools/blob/main/CONTRIBUTING.md
-->

## Summary

Closes #3962 

### noControlCharactersInRegex

description
```txt
Prevents from having control characters and some escape sequences that match control characters in regular expressions.

Control characters are hidden special characters that are numbered from 0 to 31 in the ASCII system. They're not commonly used in JavaScript text. So, if you see them in a pattern (called a regular expression), it's probably a mistake.

The following elements of regular expression patterns are considered possible errors in typing and are therefore disallowed by this rule:

- Hexadecimal character escapes from \x00 to \x1F
- Unicode character escapes from \u0000 to \u001F
- Unicode code point escapes from \u{0} to \u{1F}
- Unescaped raw characters from U+0000 to U+001F

Control escapes such as \t and \n are allowed by this rule.
```

ESLint:  https://eslint.org/docs/latest/rules/no-control-regex

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

<!-- Link any relevant issues if necessary or include a transcript of any Discord discussion. -->

## Test Plan

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output. -->
`cargo test -p rome_js_analyze -- no_control_characters_in_regex`

ematipico reviewed (commented):
I have a suggestion. Feel free to take or reject my suggestion. And if you take it, feel free to apply the suggestion in the next PR. Doing it in the next PR would allow us to catch possible regressions :)

At the moment logic of the rule creates a string every time we see a control character. I believe this is not efficient because technically we already have that string inside the node. 

My suggestion would be to store the **range** of where the control character is. With range I mean the index where the control character starts and the index where the control character ends. You can save the index as `(usize, usize)` or `TextRange`.

Doing so would allow us to do better things:
- we don't allocate a new string for every control character
- to retrieve the string, we just need do
	```rs
	let control_character = original_string[start_index..end_index]; 
	```
- we can create a `TextRange` from those indexes and show the control character in the diagnostic

unvalley commented:
@ematipico Thanks for the review! I addressed reviewed points. 
After merging this PR,  I'll create an issue for updating to use `TextRange` instead of creating new string.
